prediction_market_agent/agents/microchain_agent/blockchain/code_interpreter.py

```diff
@@ -86,10 +92,22 @@ def format_docs(docs: list[Document]) -> str:
         )
 
     def generate_summary(self, function_names: list[str]) -> Summaries:
-        function_names_formatted = "\n".join(
-            ["- " + substr for substr in function_names]
-        )
-        result: Summaries = self.rag_chain.invoke(
-            f"Create a summary for the function names listed below. Summarize, for each function, its purpose and what it does. You must produce a summary for all the functions listed below. \n  {function_names_formatted}."
-        )
-        return result
+        summaries = par_map(function_names, lambda x: self.try_summary_else_default(x))
+        return Summaries(summaries=summaries)
+
+    def try_summary_else_default(
+        self, f_name: str, default_summary: str = ""
+    ) -> FunctionSummary:
+        """
+        Tries generating a summary for the given function, else generates default description.
+        Note that try-except needed because LLM fails (sometimes) for a few functions. Alternate approaches
+        (e.g. pass in entire source code or a cleaned version) can also be explored.
+        """
+        try:
+            summary: FunctionSummary = self.rag_chain.invoke(
+                f"Create a summary for the function {f_name}."
+            )
+            return summary
+        except:
+            logger.info(f"Could not generate summary for function {f_name}")
```

Comment:
```suggestion
            logger.exception(f"Could not generate summary for function {f_name}")
```

Should be logged so we know what's going on

Reply:
I thought only `logger.info` was logged by GCP?

Reply:
`logger.debug` is not logged, otherwise everything above debug is logged

Reply:
Thanks for the info - not needed anymore since this was refactored away, hence closing.